transport_plugins/websocket/iotile_transport_websocket/virtual_websocket.py

```diff
@@ -0,0 +1,922 @@
+        before sending it to the device.
+
+        Args:
+            connection_string (str): The connection string of the device
+            chunk (str): A chunk of the script to send (up to 20 bytes), encoded using base64
+            chunk_status (tuple): Contains information as the current chunk index and the total of chunk which
+                                compose the script.
+        """
+
+        operation = operations.SEND_SCRIPT
+        connection_id = self._get_connection_id(connection_string)
+
+        if connection_id is None:
+            self.send_error(operation, 'Received script chunk from unknown connection: {}'.format(connection_string))
+            return
+
+        # Check and see if we have the entire script or if we need to accumulate it
+        index, count = chunk_status
+        if index == 0:
+            self.script = bytes()
+
+        decoded_chunk = base64.b64decode(chunk)
+        self.script += decoded_chunk
+
+        # If there is more than one chunk and we aren't on the last one, wait until we receive them
+        # all before sending them on to the device as a unit
+        if index != count - 1:
+            return
+
+        error = None
+
+        try:
+            self._simulate_send_script(connection_id, self.script, self._notify_progress)
+
+        except Exception as exc:
+            self.logger.exception('Error in manager send_script')
+            error = 'Internal error: {}'.format(str(exc))
+
+        if error:
+            self.send_error(operation, error, connection_string=connection_string)
+        else:
+            self.send_response(operation, connection_string=connection_string)
+
+    def _notify_progress(self, connection_string, done_count, total_count):
+        """Send a notification containing the current progress of the given operation. The progress is computed
+        from done_count/total_count.
+
+        Args:
+            connection_string (str): The connection string of the device where the operation is in progress
+            done_count (int): Number of chunks already processed
+            total_count (int): Number of total chunks to proceed
+        """
+
+        self.send_notification(operations.NOTIFY_PROGRESS,
+                               connection_string=connection_string,
+                               done_count=done_count,
+                               total_count=total_count)
+
+    def _stream_data(self, device_uuid, chunk=None):
+        """Stream reports to the WebSocket client in 20 byte chunks
+
+        Args:
+            device_uuid (int): The uuid of the device which sent the report
+            chunk (bytes): A chunk that should be sent instead of requesting a
+                new chunk from the pending reports.
+        """
+
+        connection_string = self._uuid_to_connection_string(device_uuid)
+
+        self.streaming_data = True
+
+        if chunk is None:
+            chunk = self._next_streaming_chunk(20)
```

Comment:
You can increase the transmit size here. The 20 bytes for the BLE virtual interface was because the notification packets used over BLE only support up to 20 bytes, here you could use your much larger MTU up to 64kb or whatever you like really.